### DISCUSSION TITLE
Using Celery with docker on 1 node VM, is it better multiple worker or big concurrent on 1 worker ?

### DISCUSSION THREAD

**Discussion by mahardiany** [apache/airflow#40678] [Q&A]
Hi Airflow community,
I want to ask about, what is better, multiple worker or 1 worker with big concurrent run process ?
In my case, I'm using Airflow with docker compose and custom Image, and I want to maximize the resource to run parallellism task.

let say, If my resource (1 node VM) can run up to 40 task run simultaneously, what is better for this architecture and why ?
1. Using 2 worker celery with each worker set to run 20 task run parallel
2. Using 1 worker celery with 40 task run parallel

And what env-variable that I need to config for this 2 option to run ?

very much appreciated for anyone that can share the experience
thnak you

nb : Im not using k8s, because I havent deep dive for k8s deployment 🙇

**Comment by potiuk** [comment]
I don't think there is a big difference

**Comment by raphaelauv** [comment]
1 node VM -> you can use localexecutor

**Comment by mahardiany** [comment]
okay thank you, i will try fot one  elery worker and localexcutor. and see which is the most optimize for my resources.
